### PR TITLE
fix: switch to eslint for Typescript linting

### PR DESCRIPTION
**Pull request by kyegupov**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
We already have switched Eslint in other projects, it's time for CLI.

### CODE REVIEW

File: src/cli/commands/auth/index.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { MisconfiguredAuthInCI } from '../../../lib/errors/misconfigured-auth-in
 import { AuthFailedError } from '../../../lib/errors/authentication-failed-error';
 import {verifyAPI} from './is-authed';
 
-module.exports = auth;
+export = auth;
```
Comment (robcresswell) [inline src/cli/commands/auth/index.ts]: Is `export =` valid syntax? Usually we use `export { auth }` etc.

Comment (kyegupov) [reply src/cli/commands/auth/index.ts]: Yes it is, and we have to use it in this case: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+module.exports = {
+  settings: { cache: true },
```
Comment (robcresswell) [inline .eslintrc.js]: Does this actually do anything? The caching is triggered by `--cache` in the `package.json`

Comment (kyegupov) [reply .eslintrc.js]: :man_shrugging: I have just copied this file from the Registry as is

Comment (robcresswell) [reply .eslintrc.js]: Huh, thats a goof on my part then, sorry. Please remove it from this PR.

File: src/cli/commands/config.js [hunk 3]
```diff
@@ -11,7 +10,7 @@ module.exports = function config(method) {
         return item.split('=');
       }).forEach((pair) => {
         res += pair[0] + ' updated\n';
-        snyk.config.set.apply(snyk.config, pair);
+        snyk.config.set(...pair);
```
Comment (dkontorovskyy) [inline src/cli/commands/config.js]: 😍

File: src/cli/commands/policy.ts [hunk 4]
```diff
@@ -7,13 +7,14 @@ async function displayPolicy(path?: string): Promise<string> {
     const loadedPolicy = await policy.load(path || process.cwd()) as Promise<string>;
     return await display(loadedPolicy);
   } catch (error) {
+      let adaptedError;
```
Comment (dkontorovskyy) [inline src/cli/commands/policy.ts]: `: CustomError` ?

File: src/lib/hotload.js [hunk 5]
```diff
@@ -8,11 +8,11 @@ const path = require('path');
 function hotload(dir) {
   return function (name) {
     let module = null;
-    return function () {
+    return function (...args) {
       if (module === null) {
         module = require(path.relative(__dirname, path.resolve(dir, name)));
       }
-      return module.apply(null, arguments);
+      return module(...args);
```
Comment (dkontorovskyy) [inline src/lib/hotload.js]: 😍

File: src/lib/isolate.js [hunk 6]
```diff
@@ -44,9 +44,9 @@ function instrumentProps(id, key, obj) {
   const original = obj;
 
   if (type === 'function') {
-    obj = function instrumented() {
+    obj = function instrumented(...args) {
       console.log('NOTIFY: %s@%s', key || id, id);
-      original.apply(this, arguments);
+      original(...args);
```
Comment (dkontorovskyy) [inline src/lib/isolate.js]: 😍

File: src/cli/commands/policy.ts (hunk 4 above)
Comment (kyegupov) [reply src/cli/commands/policy.ts]: doesn't make much difference, but why not.

**Pull request merged**